Add migration guide for v7 (#4910)
getsentry/sentry-javascript: MIGRATION.md

As far as SDK development goes, dropping support means no longer running integration tests for Node.js version 6, and also no longer handling edge cases specific to version 6.
Running the new SDK version on Node.js v6 is therefore highly discouraged.

### Removal Of Old Platform Integrations From `@sentry/integrations` Package

The following classes will be removed from the `@sentry/integrations` package and can no longer be used:

- `Angular`
- `Ember`
- `Vue`

These classes have been superseded and were moved into their own packages, `@sentry/angular`, `@sentry/ember`, and `@sentry/vue` in a previous version.
Refer to those packages if you want to integrate Sentry into your Angular, Ember, or Vue application.

### Moving To ES6 For CommonJS Files

From version 7 onwards, the CommonJS files in Sentry JavaScript SDK packages will use ES6.

If you need to support Internet Explorer 11 or old Node.js versions, we recommend using a preprocessing tool like [Babel](https://babeljs.io/) to convert Sentry packages to ES5.

### Restructuring Of Package Content

> Up until now, we have published our packages on npm with the following structure:

Reviewer: Sorry for the nit but I think it's better to be specific here. Totally optional though.

```suggestion
Up until v6.x, we have published our packages on npm with the following structure:
```

- `build` folder contained CDN bundles
- `dist` folder contained CommonJS files and TypeScript declarations
- `esm` folder contained ESM files and TypeScript declarations

Moving forward the JavaScript SDK packages will generally have the following structure:

- `cjs` folder contains CommonJS files
- `esm` folder contains ESM files
- `types` folder contains TypeScript declarations

**CDN bundles of version 7 or higher will no longer be distributed through our npm package.**
This means that most third-party CDNs like [unpkg](https://unpkg.com/) or [jsDelivr](https://www.jsdelivr.com/) will also not provide them.

> If you depend on any specific files in a Sentry JavaScript npm package, you will most likely need to update their references.

Reviewer: Should we give an example of changing an import from e.g. `@sentry/somePkg/dist/...` to `@sentry/somePkg` (recommended) or `@sentry/somePkg/cjs/..` (if really necessary)?

> ### Removing the `API` class from `@sentry/core``

Reviewer: ```suggestion
### Removing the `API` class from `@sentry/core`
```

> The internal `API` class was removed. More details can be found in the [PR that deprecated this class](https://github.com/getsentry/sentry-javascript/pull/4281). To migrate, see the following example.

Reviewer: Again an optional suggestion but for me it makes sense to specify that the functionality is not lost (if I understand the PRs correctly). 
```suggestion
The internal `API` class was removed in favour of functions. More details can be found in the [PR that deprecated this class](https://github.com/getsentry/sentry-javascript/pull/4281). To migrate, see the following example.
```
